pkg/providers/amifamily/ami.go

```diff
@@ -243,6 +242,8 @@ func GetFilterAndOwnerSets(terms []v1beta1.AMISelectorTerm) (res []FiltersAndOwn
 	idFilter := &ec2.Filter{Name: aws.String("image-id")}
 	for _, term := range terms {
 		switch {
+		case term.EKSOptimized != nil:
```

Comment:
musing: What if the Optimized AMI just mutating the selector terms in a way that it injected an additional id requirement? Then, you could just use the common function for this and wouldn't have to have separate handling here.

Consider taking this in a follow-on and backlogging it.

Reply:
Definitely an interesting idea, but I don't think we'd want to go with that specific for a couple of reasons:
- We shouldn't be calling describe-images if we don't need to. IIUC we call describe-images for ids for two reasons: to validate that the image exists and the user has access and to discover the architecture requirement. Since we already know the requirements and know that these are public AMIs this wouldn't be worth calling.
- We'd have to reinject the requirements that we know from the SSM paths after.

IMO there isn't a massive code-readability benefit and I think these drawbacks are significant enough that it's not worth pursuing. Personally, I think from a readability standpoint the only thing missing is a better collective name for the non-eksOptimized terms so we can be more specific than `getAMIs`.